presto-main/src/test/java/com/facebook/presto/operator/scalar/TestQuantileDigestFunctions.java

```diff
@@ -71,6 +72,31 @@ public void testValueAtQuantileBigint()
                 5L);
     }
 
+    @Test
+    public void testQuantileAtValueBigint()
+    {
+        QuantileDigest qdigest = new QuantileDigest(1);
+        addAll(qdigest, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9);
```

Comment:
I see some surprising results here when adding additional test cases for doubles:

* `Double.NaN` as the input value returns 1.0
* `Double.MIN_VALUE` as the input value returns 0.1

Reply:
Yes, this is expected. The Double type is not directly interfaced with the qdigest, but encoded in a special way. If you try to test the already existing value_at_quantile function, e.g., something like
```
@Test
    public void testValueAtQuantileDouble()
    {
        QuantileDigest qdigest = new QuantileDigest(1);
        addAll(qdigest, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9);
        functionAssertions.assertFunction(format("value_at_quantile(CAST(X'%s' AS qdigest(double)), 0.5)", toHexString(qdigest)),
                DOUBLE,
                5L);
    }
```
you will find that the output is 2.5E-323, not 5.0

Reply:
Suppose you were to wrap each of those integer literals into `doubleToSortableLong`, as the QDigest aggregation function does, you'll still see this result.

Reply:
I tried this example:
```
@Test
    public void testQuantileAtValueDouble()
    {
        QuantileDigest qdigest = new QuantileDigest(1);
        addAll(qdigest, doubleToSortableLong(0.0),doubleToSortableLong(1),doubleToSortableLong(2),doubleToSortableLong(3),doubleToSortableLong(4),doubleToSortableLong(5),doubleToSortableLong(6),doubleToSortableLong(7),doubleToSortableLong(8),doubleToSortableLong(9));
        functionAssertions.assertFunction(format("quantile_at_value(CAST(X'%s' AS qdigest(double)), 6)", toHexString(qdigest)),
                DOUBLE,
                0.6);
    }
```
And it gives me the right output 0.6 as expected.

Reply:
Yes, that's expected.  However, if you try to replace `0.6` with `nan()` in the assert above, it will return 1.0--this is strange because it implies `NaN` is greater than all of the values in the digest.

Reply:
Ok, I see you point. I'm not quite familiar with the internals of QDigest data structure, do you have any suggestion on what is the min/max range for it? I have a rough guess of [-2^63,  2^63 - 1] ?

Reply:
There is a `getMax` and `getMin` on the data structure which will tell you what is the max/min for the specific one you are using.

Reply:
Added checking on current min/max range of the QDigest, and will return null when the input value is beyond the range. And also added the nan() test case in the unit test, and verified it is now returning null.